src/agent/coverage/src/block/linux.rs

```diff
@@ -40,7 +41,14 @@ impl<'c> Recorder<'c> {
         let mut tracer = Ptracer::new();
         let mut child = tracer.spawn(cmd)?;
 
-        let _timer = Timer::new(timeout, move || child.kill());
+        let timed_out = Arc::new(AtomicBool::new(false));
+
+        let timer_flag = timed_out.clone();
```

Comment:
Maybe this could be internal to the `Timer` struct, and set right before we call [`on_timeout()`](https://github.com/microsoft/onefuzz/blob/b3fd6d5ed0fcc851478a02f169e0f5ae70f114a6/src/agent/coverage/src/block/linux.rs#L458-L459)?

Then we could add a `timed_out(&self) -> bool` method to `Timer` to further hide the atomics.

Reply:
At the moment I have it as conditional on whether or not the kill actually "worked" so that we don't report an error if we timeout but the process had already stopped, so the `timeout` function will need to return some kind of result to move that logic into `Timer`.

Reply:
Alternately: if the process _would_ have exited on its own, there's always still a race with our call to `kill()`. Checking the `kill()` result for an `InvalidInput` case doesn't squash the race.

So, I think we could fairly just say that if we made it to the `on_timeout()` callback at all, the timeout state was semantically reached (even if we could have in principle observed the exit).